pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_zone_config_system_tenant failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_zone_config_system_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/63d14e26-25b4-4ef6-b012-c175658024c2?testReportRun=3&testReportShard=15&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on release-24.1 @ [557f4e6fe89d683d49082d88ba5adf6660e8f16f](https://github.com/cockroachdb/cockroach/commits/557f4e6fe89d683d49082d88ba5adf6660e8f16f):


```
	NOTICE: .NodeID() called via implicit interface StorageLayerInterface;
HINT: consider using .StorageLayer().NodeID() instead.

pkg/testutils/physicalplanutils/fake_resolver.go:28: (FakeResolverForTestCluster)
	NOTICE: .AdvRPCAddr() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().AdvRPCAddr() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
[09:24:51] setting distsql_workmem='33410B';
    logic.go:4322: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_zone_config_system_tenant861499366
--- FAIL: TestLogic_zone_config_system_tenant (68.68s)
=== RUN   TestLogic_zone_config_system_tenant/dropped_relation_gc_ttl
[09:24:56] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 30 statements
[09:24:59] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 31 statements
[09:25:01] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 33 statements
    logic.go:2963: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant:187: SELECT
          crdb_internal.pretty_key(start_key, -1),
          crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)->'gcPolicy'->>'ttlSeconds'
        FROM system.span_configurations
        WHERE start_key >= (SELECT crdb_internal.table_span(100)[1])
        ORDER BY start_key
        expected:
            /Table/106  14400
            /Table/107  14400
            /Table/110  90001
            /Table/111  90001
            /Table/112  14400
            
        but found (query options: "retry") :
            /Table/106  14400
            /Table/107  14400
            /Table/110  90001
            /Table/111  1
            /Table/112  1
            
[09:25:46] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 34 statements
[09:25:46] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant with config fakedist-disk: 34 tests, 1 failures
[09:25:52] --- total progress: 34 statements
--- total: 34 tests, 1 failures
    --- FAIL: TestLogic_zone_config_system_tenant/dropped_relation_gc_ttl (53.38s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=15</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_zone_config_system_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37210

--- a/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
+++ b/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
@@ -194,8 +194,8 @@ ORDER BY start_key
 /Table/106  14400
 /Table/107  14400
 /Table/110  90001
-/Table/111  90001
-/Table/112  14400
+/Table/111  1
+/Table/112  1
 
 subtest transactional_schemachanges
 

